### PR TITLE
add UseEmulator api to auth

### PR DESCRIPTION
**Pull request by cynthiajoan**
1. Add UseEmulator api to auth
2. Remove obsolete heartbeat blocks.
***

[replace this line]: # (Describe your testing in detail.)
***

### Type of Change
Place an `x` the applicable box:
- [ ] Bug fix. Add the issue # below if applicable.
- [ ] New feature. A non-breaking change which adds functionality.
- [ ] Other, such as a build process or documentation change.
***

#### Notes
- Bug fixes and feature changes require an update to the `Release Notes` section of `release_build_files/readme.md`.
- Read the contribution guidelines [CONTRIBUTING.md](https://github.com/firebase/firebase-cpp-sdk/blob/main/CONTRIBUTING.md).
- Changes to the public API require an internal API review. If you'd like to help us make Firebase APIs better, please propose your change in a feature request so that we can discuss it together.

### CODE REVIEW

File: auth/src/include/firebase/auth.h [hunk 1]
```diff
@@ -631,6 +632,14 @@ class Auth {
   void RemoveIdTokenListener(IdTokenListener* listener);
 #endif  // not SWIG
 
+#if !defined(DOXYGEN)
```
Comment (a-maurice) [inline auth/src/include/firebase/auth.h:635]: Should probably hide from SWIG for now (unless the plan is for this to also be visible from Unity)

File: auth/src/include/firebase/auth.h [hunk 2]
```diff
@@ -631,6 +632,14 @@ class Auth {
...
+  ///
+  /// Modify this Auth instance to communicate with the Firebase Authentication
+  /// emulator.
+  ///
+  void useEmulator(std::string host, uint32_t port);
```
Comment (a-maurice) [inline auth/src/include/firebase/auth.h:640]: "UseEmulator", with a capital U?

File: auth/src/desktop/rpcs/auth_request.cc [hunk 3]
```diff
@@ -33,54 +35,22 @@ const char* kHeaderFirebaseLocale = "X-Firebase-Locale";
...
 AuthRequest::AuthRequest(::firebase::App& app, const char* schema,
                          bool deliver_heartbeat)
-    : RequestJson(schema) {
-  // The user agent strings are cached in static variables here to avoid
-  // dependencies upon other parts of this library.  This complication is due to
-  // the way the tests are currently configured where each library has minimal
-  // dependencies.
-
-  CheckEmulator();
-  static std::string auth_user_agent;           // NOLINT
```
Comment (a-maurice) [inline auth/src/desktop/rpcs/auth_request.cc:43]: I'm guessing the user_agent stuff should not be getting removed

Comment (cynthiajoan) [reply auth/src/desktop/rpcs/auth_request.cc:43]: I consulted with Shawn and they are actually not needed anymore. Include Shawn to review to confirm

File: auth/src/include/firebase/auth.h (hunk 1 above)
Comment (cynthiajoan) [reply auth/src/include/firebase/auth.h]: done

File: auth/src/include/firebase/auth.h (hunk 2 above)
Comment (cynthiajoan) [reply auth/src/include/firebase/auth.h]: done

**Review by a-maurice — APPROVED** [review]

File: auth/src/desktop/rpcs/auth_request.cc [hunk 4]
```diff
@@ -33,54 +35,22 @@ const char* kHeaderFirebaseLocale = "X-Firebase-Locale";
...
-        app.GetHeartbeatController();
-    if (heartbeat_controller) {
-      std::string payload = heartbeat_controller->GetAndResetStoredHeartbeats();
-      std::string gmp_app_id = app.options().app_id();
-      if (!payload.empty()) {
-        add_header(app_common::kApiClientHeader, payload.c_str());
-        add_header(app_common::kXFirebaseGmpIdHeader, gmp_app_id.c_str());
-      }
-    }
-  }
```
Comment (chkuang-g) [inline auth/src/desktop/rpcs/auth_request.cc:80]: This part should keep. This is for the new desktop heartbeat Matt implemented.

File: auth/src/desktop/rpcs/auth_request.cc (hunk 3 above)
Comment (chkuang-g) [reply auth/src/desktop/rpcs/auth_request.cc:43]: Most of the user-agent stuffs are not used, at least by Platform Logging pipeline, at all, except for the section I commented below. Therefore it is safe to remove.

However, I wonder if this should be in this particular PR. 

If you are planning to bundle this change along with emulator support, please make sure you include this information in the commit message, so it does not look like an accident change.

**Review by chkuang-g — CHANGES_REQUESTED** [review]

File: auth/src/desktop/rpcs/auth_request.cc (hunk 3 above)
Comment (cynthiajoan) [reply auth/src/desktop/rpcs/auth_request.cc:43]: done

File: auth/src/desktop/rpcs/auth_request.cc (hunk 4 above)
Comment (cynthiajoan) [reply auth/src/desktop/rpcs/auth_request.cc]: done

**Review by chkuang-g — APPROVED** [review]

**Pull request merged**